core/kazoo_documents-1.0.0/src/kzd_agent.erl

```diff
@@ -0,0 +1,40 @@
+%%%-------------------------------------------------------------------
+%%% @copyright (C) 2010-2015, 2600Hz
+%%% @doc
+%%%
+%%% @end
+%%% @contributors
+%%%   SIPLABS, LLC (Maksim Krzhemenevskiy)
+%%%-------------------------------------------------------------------
+-module(kzd_agent).
+
+-include("kz_documents.hrl").
+
+%% API
+-export([maybe_add_queue/2, maybe_add_queue/3
+         ,maybe_rm_queue/2, maybe_rm_queue/3
+        ]).
+
+-spec maybe_add_queue(wh_json:object(), ne_binary()) -> wh_json:object().
+maybe_add_queue(AgentJObj, QueueId) ->
+    maybe_add_queue(AgentJObj, QueueId, AgentJObj).
+
+-spec maybe_add_queue(wh_json:object(), ne_binary(), term()) -> wh_json:object() | term().
+maybe_add_queue(AgentJObj, QueueId, WhenExists) ->
+    Qs = wh_json:get_value(<<"queues">>, AgentJObj, []),
+    case lists:member(QueueId, Qs) of
+        'false' -> wh_json:set_value(<<"queues">>, [QueueId | Qs], AgentJObj);
+        'true' -> WhenExists
+    end.
+
+-spec maybe_rm_queue(wh_json:object(), ne_binary()) -> wh_json:object().
+maybe_rm_queue(AgentJObj, QueueId) ->
+    maybe_rm_queue(AgentJObj, QueueId, AgentJObj).
+
+-spec maybe_rm_queue(wh_json:object(), ne_binary(), term()) -> wh_json:object() | term().
```

Comment:
You can use `Default` instead of `term()` in the spec, so Dialyzer knows its either the JSON object or whatever `Default` is. The union of `wh_json:object() | term()` is `term()` which isn't too helpful. Does that make sense?